enhancements/operator-framework-api/catalogue-check.md

```diff
@@ -0,0 +1,377 @@
+
+## Drawbacks
+
+We might use this scenario as use case to know how we will address this kind of requirement only for downstream in the future.
+
+However, if we check that the effort/cost to address this needs in the possible agreed design will be too high which not justify its value at the moment then, we might decide to not move forward with now.
+
+### Test Plan
+
+The rules implemented should be covered by unit tests.
+
+## Alternatives
+
+### (Option A) Include the OpenShift Validator in upstream
+
+The proposal above was defined considering the new Validator be implemented in [operator-framework/api][oper-api] and used by SDK upstream implementation.
+
+**Pros**
+
+- Lower effort/cost since it would be similar to the others as it usage
+- Allow the Validator be used with OKD
+- The Validator can be used by any solution
+- Does not requires the usage/installation of a downstream SDK binary
+
+**NOTE** SDK users and Pipeline uses the upstream binary and shows that we have not a release process for SDK binary from downstream. Just for the images which are useful for Ansible/Helm operators.
+
+- Provide specific checks for OpenShift by default can help users which are from upstream and would like to publish their solutions on OperatorHub and OpenShift.
+- Users will still able to get the latest releases and the bug fixes faster and will not be affected for any delay that might can be faced to get the solutions from downstream.
+- Provide and promote the helper to the users since they can check that it is available by using the `--list-optional` flag:
+
+```sh
+$ operator-sdk bundle validate --list-optional
+NAME           LABELS                     DESCRIPTION
+operatorhub    suite=operatorframework    OperatorHub.io metadata validation
+               name=operatorhub          
+```
+
+**Cons**
+
+- Provide specific OpenShift solutions in upstream might hurt the community
+- OLM team would not like to add a code which is valid only for OpenShift unless we provide others vendor's validators as well in the public API. They are trying decouple the rules which are specific for OCP from OLM.
+- In the future it might result an effort required to move any OpenShift rules for downstream repos and implementations which are only valid for it. Currently, it would be the only case scenario so far, however, the precedence can bring others ones.
+
+### (Option B) Include the OpenShift Validator in downstream component for [operator-framework/api][oper-api] and use it in SDK upstream repository
+
+**Pros**
+
+- Mainly all pros of option A
+- Solves the OLM team concern about add OpenShift checks in [operator-framework/api][oper-api]
+  
+**Cons**
+
+- Effort required to keep a downstream repository for a downstream component and its releases [operator-framework/api][oper-api]
+- SDK would need to probably use only the downstream import to avoid misleading
```

Comment:
SDK upstream would NOT be using a downstream o-f/api. So the con is that anyone that wants to target openshift *MUST* use a downstream SDK. That's actually pretty bad because today it honestly doesn't matter which one you use.

If we must do Option B, then there *MUST* be a mechanism to dynamically load these downstream rules into an upstream at runtime.

Reply:
Could we not import both (downstream and upstream) APIS as suggested here in the SDK repo?